src/Models/HydrostaticFreeSurfaceModels/calculate_hydrostatic_free_surface_tendencies.jl

```diff
@@ -143,3 +142,46 @@ end
     i, j = @index(Global, NTuple)
     @inbounds Gη[i, j, 1] = free_surface_tendency(i, j, grid, args...)
 end
+
+#####
+##### Boundary condributions to hydrostatic free surface model
+#####
+
+""" Apply boundary conditions by adding flux divergences to the right-hand-side. """
+function calculate_hydrostatic_boundary_tendency_contributions!(Gⁿ, arch, velocities, free_surface, tracers, clock, model_fields)
+
+    barrier = Event(device(arch))
+
+    events = []
+
+    # Velocity fields
+    for i in (:u, :v)
```

Comment:
This looks like it will work but considering that we are doing essentially the same thing for the fields, `u,v,\eta,tracers`, I do wonder whether this could be rewrittein as a loop over the fields.  I don't believe this would improve the functionality at all, but might make the code a bit easier to read.  Just a suggestion.

Reply:
It's a fair point --- we could combine all the `boundary_tendencies` functions into one!

Reply:
Since you are keen to merge this then maybe better to save this for a futrue PR